orangecontrib/argument/miner/chunker.py

```diff
@@ -23,170 +19,273 @@
+    p_scores: List[float],
+    topics: List[int],
+    ranks: List[float],
+):
+    """Given all the measures of chunks, generate and return the chunk table as a pandas dataframe, with pre-defined column names.
+
+    Args:
+        arg_ids (List[int]): ids of arguments that chunks belong to
+        chunks (List[str]): chunk text
+        p_scores (List[float]): polarity score of chunks
+        topics (List[int]): topic id of chunks
+        ranks (List[float]): rank of chunks
+
+    Returns:
+        pd.DataFrame: chunk table
     """
-    BERT-based topic modeling and doc ranking.
+    return pd.DataFrame(
+        {
+            "argument_id": arg_ids,
+            "chunk": chunks,
+            "polarity_score": p_scores,
+            "topic": topics,
+            "rank": ranks,
+        }
+    )
+
+
+class TopicModel:
+    """Topic modeling class.
+
+    Functions are implemented based on the BERTopic model. For now, the topic model is setup with a set of default parameters of the sub-models. However, it should be possible that the user can config it further. This will be a next step.
+
+    Attributes:
+        setup (dict): setup of the sub-models. Default values::
+            {
+                "language": "english"
+                "transformer: "all-mpnet-base-v1",
+                "n_components": 5,
+                "min_cluster_size": 10,
+                "ngram_range": [1, 1]
+            }
+        embed_model (:obj:'SentenceTransformer'): instance of  the sentence transformer as the embedding sub-model.
+        rd_model (:obj:'UMAP'): instance of UMAP algorithm as the dimensionality reduction sub-model.
+        cluster_model (:obj:'HDBSCAN'): instance of HDBSCAN as the clustering sub-model.
+        vector_model (:obj:'CountVectorizer'): instance of CountVectorizer as the vectorization sub-model.
+        ctfidf_model (:obj:'ClassTfidfTransformer'): instance of ClassTfidfTransformer as the ctfidf sub-model.
+        represent_model (:obj:'PartOfSpeech'): instance of PartOfSpeech as the topic representation sub-model.
+        model (:obj:'BERTopic'): the topic model that applied the sub-models predefined.
     """
 
     def __init__(self):
-        self.embedding_model = SentenceTransformer("all-mpnet-base-v1")
-        self.umap_model = UMAP(n_components=5)
-        self.hdbscan_model = HDBSCAN(min_cluster_size=10, prediction_data=True)
-        self.vectorizer_model = CountVectorizer(
-            stop_words="english", ngram_range=[1, 1]
+        self.setup = {
+            "language": "english",
+            "transformer": "all-mpnet-base-v1",
+            "n_components": 5,
+            "min_cluster_size": 5,
+            "ngram_range": [1, 1],
+        }
+        self.embed_model = None
+        self.rd_model = None
+        self.cluster_model = None
+        self.vector_model = None
+        self.ctfidf_model = None
+        self.represent_model = None
+        self.model = None
+
+    def init_models(self):
```

Comment:
I am not sure why `init_models()` is a separate method. It looks to me like it conceptually belongs to the constructor, resulting in calls like this one from above:
```python
    topic_model = TopicModel()
    topic_model.init_models()
```
Is there a scenario in which you would construct a `TopicModel` object, but not call `init_models()` on it? If not, I would call `self.init_models()` from the `__init__()` method, and possibly make it a private method (or even merge it into `__init__()`) for conciseness and to prevent user mistakes.

Reply:
I added a line to call `init_model` in the constructor to avoid a separate call when instantiating `TopicModel`. However, I would still prefer to keep this method, as there is a (potential) scenario of updating the parameters of the sub-models. With `init_model`, there is no need to create a completely new instance of `TopicModel`.